contracts/asset-proxy/contracts/src/bridges/DydxBridge.sol

```diff
@@ -0,0 +1,186 @@
+/*
+
+  Copyright 2019 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-utils/contracts/src/DeploymentConstants.sol";
+import "../interfaces/IERC20Bridge.sol";
+import "../interfaces/IDydxBridge.sol";
+import "../interfaces/IDydx.sol";
+
+
+contract DydxBridge is
+    IERC20Bridge,
+    IDydxBridge,
+    DeploymentConstants
+{
+
+    /// @dev Callback for `IERC20Bridge`. Deposits or withdraws tokens from a dydx account.
+    ///      Notes:
+    ///         1. This bridge must be set as an operator of the input dydx account.
+    ///         2. This function may only be called in the context of the 0x Exchange.
+    ///         3. The maker of the 0x order must be the dydx account owner.
+    ///         4. To deposit into dydx use this function in the `takerAssetData`.
+    ///            (The `to` address is the 0x order maker and dydx account owner).
+    ///         5. To withdraw from dydx use this function in the `makerAssetData`.
+    ///            (The `from` address is the 0x order maker and dydx account owner).
+    /// @param from The sender of the tokens.
+    /// @param to The recipient of the tokens.
+    /// @param amount Minimum amount of `toTokenAddress` tokens to deposit or withdraw.
+    /// @param encodedBridgeData An abi-encoded `BridgeData` struct.
+    /// @return success The magic bytes if successful.
+    function bridgeTransferFrom(
+        address,
+        address from,
+        address to,
+        uint256 amount,
+        bytes calldata encodedBridgeData
+    )
+        external
+        returns (bytes4 success)
+    {
+        // Ensure that only the `ERC20BridgeProxy` can call this function.
+        require(
+            msg.sender == _getERC20BridgeProxyAddress(),
+            "DydxBridge/ONLY_CALLABLE_BY_ERC20_BRIDGE_PROXY"
+        );
+
+        // Decode bridge data.
+        (BridgeData memory bridgeData) = abi.decode(encodedBridgeData, (BridgeData));
+
+        // Cache dydx contract.
+        IDydx dydx = IDydx(_getDydxAddress());
+
+        // Create dydx action.
+        IDydx.AccountInfo[] memory accounts = new IDydx.AccountInfo[](1);
+        IDydx.ActionArgs[] memory actions = new IDydx.ActionArgs[](1);
+        if (bridgeData.action == BridgeAction.Deposit) {
```

Comment:
Actually, how would a deposit work? Wouldn't the bridge proxy revert because no tokens were transferred?

Reply:
Hmm that's a good point. This seems to break the rules of the proxy. Can we write a test for this case before merging?

Reply:
I've reverted to the previous design iteration, which always used `from` as the account owner and allowed actions to be bundled. It's a bit more complex than this one, but it's very flexible and should address both of your comments @dorothy-zbornak.

The 0x Order in this design looks like this:
```
{
...
takerAssetData: erc20Transfer()
makerAssetData: bridgeTransfer([deposit to dydx from maker, withdraw from dydx to taker])
}
```

Reply:
Worth noting that the motivation for this ^ design was to make the bridge flexible and extensible. This allows us to iterate on the impl and experiment with other dydx actions, like executing callbacks. It also allows one 0x order to execute across several of the maker's dydx accounts. 

The alternative would be to have a single workflow that just does a basic deposit & withdrawal. I'm open to this bc it's simpler, but I do like the flexible version. Am looking forward to playing with it irl and potentially finding new use cases.